### PR TITLE
pluma-time-plugin: Use g_date_time_format

### PR DESCRIPTION
**Pull request by rbuj**
based on https://github.com/GNOME/gedit/commit/121f02a3ea066565eeae5dd42868f822ecce4924 and https://github.com/GNOME/gedit/commit/ef65fda074d58296b8c919f64744e48a1e610123

### CODE REVIEW

**Comment by rbuj** [comment]
I think i have a deja vu...

**Comment by rbuj** [comment]
A rewritten function that does not use tabs for indent, where is the code formatting issue?

**Comment by sc0w** [comment]
> A rewritten function that does not use tabs for indent, where is the code formatting issue?

It is ok for you, not for me. Tabs or spaces for indent in **all** the file is the right way for me.

Thanks for new commit :)

**Comment by raveit65** [comment]
Hmm, for me the last commit isn't  necessary.
I mean replacing tabs with spaces is Ok , but starting a line with 8 spaces is really wasteful.
And last commit has indent issues.
Using only the first commit is OK for me.....
....or you fix the indent problems.

File: plugins/time/pluma-time-plugin.c [hunk 1]
```diff
@@ -177,227 +177,227 @@ static void time_cb (GtkAction *action, PlumaTimePlugin *plugin);
...
-						 GETTEXT_PACKAGE);
-	gtk_action_group_add_actions (data->action_group,
-				      	   action_entries,
-				      	   G_N_ELEMENTS (action_entries),
-				           plugin);
+        data->action_group = gtk_action_group_new ("PlumaTimePluginActions");
+        gtk_action_group_set_translation_domain (data->action_group,
+                                                 GETTEXT_PACKAGE);
+        gtk_action_group_add_actions (data->action_group,
+                                           action_entries,
```
Comment (raveit65) [inline plugins/time/pluma-time-plugin.c:275]: Indents..........or delete this commit ;)

File: plugins/time/pluma-time-plugin.c [hunk 2]
```diff
@@ -177,227 +177,227 @@ static void time_cb (GtkAction *action, PlumaTimePlugin *plugin);
...
 /* whether we should prompt the user or use the specified format */
 static PlumaTimePluginPromptType
 get_prompt_type (PlumaTimePlugin *plugin)
 {
-	PlumaTimePluginPromptType prompt_type;
+        PlumaTimePluginPromptType prompt_type;
 
-	prompt_type = g_settings_get_enum (plugin->priv->settings,
-			        	       PROMPT_TYPE_KEY);
+        prompt_type = g_settings_get_enum (plugin->priv->settings,
```
Comment (raveit65) [inline plugins/time/pluma-time-plugin.c:326]: Indents..........or delete this commit ;)

File: plugins/time/pluma-time-plugin.c [hunk 3]
```diff
@@ -177,227 +177,227 @@ static void time_cb (GtkAction *action, PlumaTimePlugin *plugin);
...
-					   PROMPT_TYPE_KEY))
-	{
-		return;
-	}
-
-	g_settings_set_enum (settings,
-				 PROMPT_TYPE_KEY,
-				 prompt_type);
+        if (!g_settings_is_writable (settings,
+                                           PROMPT_TYPE_KEY))
```
Comment (raveit65) [inline plugins/time/pluma-time-plugin.c:337]: Indents..........or delete this commit ;)

File: plugins/time/pluma-time-plugin.c [hunk 4]
```diff
@@ -177,227 +177,227 @@ static void time_cb (GtkAction *action, PlumaTimePlugin *plugin);
...
+        {
+                return;
+        }
+
+        g_settings_set_enum (settings,
+                                 PROMPT_TYPE_KEY,
```
Comment (raveit65) [inline plugins/time/pluma-time-plugin.c:343]: Indents..........or delete this commit ;)

File: plugins/time/pluma-time-plugin.c [hunk 5]
```diff
@@ -177,227 +177,227 @@ static void time_cb (GtkAction *action, PlumaTimePlugin *plugin);
...
 static gchar *
 get_selected_format (PlumaTimePlugin *plugin)
 {
-	gchar *sel_format;
+        gchar *sel_format;
 
-	sel_format = g_settings_get_string (plugin->priv->settings,
-					      SELECTED_FORMAT_KEY);
+        sel_format = g_settings_get_string (plugin->priv->settings,
+                                              SELECTED_FORMAT_KEY);
```
Comment (raveit65) [inline plugins/time/pluma-time-plugin.c:354]: Indents..........or delete this commit ;)

File: plugins/time/pluma-time-plugin.c [hunk 6]
```diff
@@ -177,227 +177,227 @@ static void time_cb (GtkAction *action, PlumaTimePlugin *plugin);
...
 
-	return sel_format ? sel_format : g_strdup (formats [0]);
+        return sel_format ? sel_format : g_strdup (formats [0]);
 }
 
 static void
 set_selected_format (GSettings *settings,
-		     const gchar     *format)
+                     const gchar     *format)
```
Comment (raveit65) [inline plugins/time/pluma-time-plugin.c:361]: Indents..........or delete this commit ;)

Ok, i see there are more issues.
This commit doesn't make any sense for me....
I am still surprise that someone could approve that........

**Comment by rbuj** [comment]
> Hmm, for me the last commit isn't necessary.

I removed the last commit.

**Comment by raveit65** [comment]
I will fix formatting in pluma after i did it for mate-applets.

**Review by raveit65 — APPROVED** [review]
LGTM.
The time-plugin works same like before..

**Pull request merged**